docs/architecture/README.md

```diff
@@ -26,86 +26,88 @@ More detail on configurations are available [here](../administrator-guide/config
+Since the application is a Kafka Stream, it can be scaled infinitely (within the limits of partitions count on Kafka), but since no heavy computations are done in the Executor, this server component does not require a lot of resources (unless you have a very high rate of executions).
 
 
 ### Worker
-A worker is where the task is done. A Worker will receive a task from the Executor and will process it. Since tasks can be nearly anything(heavy computations, simple api calls, etc.), the worker is essentially a ThreadPool where you can configure the amount of threads you need.
+A Worker is where the task is run. A Worker will receive a task from the Executor and will process it. Since tasks can be nearly anything (heavy computations, simple api calls, etc...), the Worker is essentially a Thread Pool where you can configure the amount of threads you need.
 
-Internally, workers are a simple [Kafka Consumer](https://kafka.apache.org/documentation/#consumerapi) that will process any Task Run submitted. Workers will receive all tasks and dispatch internally in its Thread Pool.
+Internally, Workers are a simple [Kafka Consumer](https://kafka.apache.org/documentation/#consumerapi) that will process any Task Run submitted to it. Workers will receive all tasks and dispatch them internally in its Thread Pool.
 
-You can also scale workers as necessary, and have many instances on multiple servers (within the limits of the partition count on Kafka), each with its own Thread Pool. If a worker is dead, the Executor will detect it and resubmit their current taskrun to another worker.
+You can also scale Workers as necessary, and have many instances on multiple servers (within the limits of the partitions count on Kafka), each with its own Thread Pool. If a Worker is dead, the Executor will detect it and resubmit their current task run to another Worker.
 
 ### Scheduler
-Scheduler will handle most of the [triggers](../developer-guide/triggers) (except the [Flow Trigger handle by executor](../developer-guide/triggers/flow.md)). It will continuously watch all triggers and detect if all the conditions are met to trigger an Execution (submitted to Executor).
+The Scheduler will handle most of the [Triggers](../developer-guide/triggers) (except the [Flow Trigger handle by the Executor](../developer-guide/triggers/flow.md)). It will continuously watch all triggers and, if all the conditions are met, will trigger an Execution (submitted to the Executor).
 
-Internally, a Scheduler has two ThreadPools:
+Internally, a Scheduler has two Thread Pools:
 
-- **ScheduledExecutorService** : This Threadpool will tick every second and try to find an available job.
-- **CachedThreadPool** : This Threadpool will compute in a separate Thread for all the triggers.
+- **ScheduledExecutorService**: This Thread Pool will tick every second and try to find an available job.
+- **CachedThreadPool**: This Thread Pool will compute in a separate Thread for all the triggers.
```

Comment:
@tchiotludo this sentence seems strange. What will be computed by this thread pool ?

Reply:
Clearly, this is not clear here :zipper_mouth_face: 
The thread pool will do long polling for all except the schedule: detect files, execute query and wait for result, fetch a queue and trigger execution when there is result. 
Maybe the word compute is not the good one

Reply:
Thanks, more clear now. I'll send a small PR.